### PR TITLE
Raycast

### PR DESCRIPTION
**Pull request by r0xly**
Every function that used FindPartOnRay now uses workspace:Raycast

### CODE REVIEW

File: src/ReplicatedStorage/Modules/Effects/Explosions/Mob/PhyscoRocks.lua
Comment (cyrus01337) [inline src/ReplicatedStorage/Modules/Effects/Explosions/Mob/PhyscoRocks.lua:1]: This is fine in most scenarios, if this causes an issue with how a projectile is angled then I will hotfix the direction myself. Here's to hoping that won't be the case.

File: src/ReplicatedStorage/Modules/Effects/VfxHandler/FloorFreeze.lua [hunk 1]
```diff
@@ -7,6 +7,7 @@ local PhysicsService = game:GetService("PhysicsService")
 
 local Debris = game:GetService("Debris")
 local TweenService = game:GetService("TweenService")
+local Workspace = game:GetService("Workspace")
```
Comment (cyrus01337) [inline src/ReplicatedStorage/Modules/Effects/VfxHandler/FloorFreeze.lua]: Prefer `workspace` over the Workspace service for consistency.

File: src/ReplicatedStorage/Modules/Effects/VfxHandler/Rockszsz.lua [hunk 2]
```diff
@@ -3,6 +3,7 @@ local ReplicatedStorage = game:GetService("ReplicatedStorage")
 local TweenService = game:GetService("TweenService")
 local RunService = game:GetService("RunService")
 local Debris = game:GetService("Debris")
+local Workspace = game:GetService("Workspace")
```
Comment (cyrus01337) [inline src/ReplicatedStorage/Modules/Effects/VfxHandler/Rockszsz.lua]: Prefer `workspace` over the Workspace service for consistency.

File: src/ServerScriptService/Server/Characters/Aokiji/init.lua [hunk 3]
```diff
@@ -8,6 +8,7 @@ local ServerScriptService = game:GetService("ServerScriptService")
 local Debris = game:GetService("Debris")
 local RunService = game:GetService("RunService")
 local TweenService = game:GetService("TweenService")
+local Workspace = game:GetService("Workspace")
```
Comment (cyrus01337) [inline src/ServerScriptService/Server/Characters/Aokiji/init.lua]: Prefer `workspace` over the Workspace service for consistency.

File: src/StarterPlayer/StarterPlayerScripts/Client/Core/VFX/Characters/AokijiVFX.lua [hunk 4]
```diff
@@ -60,8 +60,14 @@ end
...
 function GetMousePos(X,Y,Z,Boolean)
 	local RayMag1 = workspace.CurrentCamera:ScreenPointToRay(X, Y) 
-	local NewRay = Ray.new(RayMag1.Origin, RayMag1.Direction * ((Z and Z) or 200))
-	local Target,Position,Surface = workspace:FindPartOnRayWithIgnoreList(NewRay, {Character,workspace.World.Visuals})
+
+	local RayParam = RaycastParams.new()
+	RayParam.FilterType = Enum.RaycastFilterType.Exclude
+	RayParam.FilterDescendantsInstances = { Character, workspace.World.Visuals }
+
+	local RaycastResult = workspace:Raycast(RayMag1.Origin, RayMag1.Direction * ((Z and Z) or 200))
```
Comment (cyrus01337) [inline src/StarterPlayer/StarterPlayerScripts/Client/Core/VFX/Characters/AokijiVFX.lua]: Could you explain why `(Z and Z)` is there?

**Review by cyrus01337 — CHANGES_REQUESTED** [review]

File: src/StarterPlayer/StarterPlayerScripts/Client/Core/VFX/Characters/AokijiVFX.lua (hunk 4 above)
Comment (r0xly) [reply src/StarterPlayer/StarterPlayerScripts/Client/Core/VFX/Characters/AokijiVFX.lua]: Typo. Will fix.

**Review by cyrus01337 — APPROVED** [review]

**Pull request merged**